catalyst/dl/experiment/gan.py

```diff
@@ -0,0 +1,158 @@
+                to the console.
+            state_kwargs (dict): additional state params to ``RunnerState``
+            checkpoint_data (dict): additional data to save in checkpoint,
+                for example: ``class_names``, ``date_of_training``, etc
+            distributed_params (dict): dictionary with the parameters
+                for distributed and FP16 method
+            monitoring_params (dict): dict with the parameters
+                for monitoring services
+            initial_seed (int): experiment's initial seed value
+            phase2callbacks (dict): dictionary with lists of callback names
+                which should be wrapped for appropriate phase
+                for example: {"generator_train": "loss_g", "optim_g"}
+                "loss_g" and "optim_g" callbacks from callbacks dict
+                will be wrapped for "generator_train" phase
+                in wrap_callbacks method
+        """
+        super().__init__(
+            model,
+            loaders,
+            callbacks,
+            logdir,
+            stage,
+            criterion,
+            optimizer,
+            scheduler,
+            num_epochs,
+            valid_loader,
+            main_metric,
+            minimize_metric,
+            verbose,
+            state_kwargs,
+            checkpoint_data,
+            distributed_params,
+            monitoring_params,
+            initial_seed,
+        )
+        self.wrap_callbacks(phase2callbacks)
+
+    def wrap_callbacks(self, phase2callbacks) -> None:
+        """Phase wrapping procedure for callbacks"""
+        discriminator_phase_name = self._additional_state_kwargs[
+            "discriminator_train_phase"
+        ]
+        discriminator_phase_num = self._additional_state_kwargs[
+            "discriminator_train_num"
+        ]
+        generator_phase_name = self._additional_state_kwargs[
+            "generator_train_phase"
+        ]
+        generator_phase_num = self._additional_state_kwargs[
+            "generator_train_num"
+        ]
+        self._callbacks["phase_manager"] = PhaseManagerCallback(
+            train_phases=OrderedDict(
+                [
+                    (discriminator_phase_name, discriminator_phase_num),
+                    (generator_phase_name, generator_phase_num),
+                ]
+            ),
+            valid_mode="all",
+        )
+        for phase_name, callback_name_list in phase2callbacks.items():
+            # TODO: Check for phase in state_params?
+            for callback_name in callback_name_list:
+                callback = self._callbacks.pop(callback_name)
+                self._callbacks[callback_name] = PhaseBatchWrapperCallback(
+                    base_callback=callback, active_phases=[phase_name],
+                )
+
+    def get_callbacks(self, stage: str) -> "OrderedDict[str, Callback]":
+        callbacks = super().get_callbacks(stage="")
```

Comment:
why is `stage=""`?

Reply:
No default param value for this argument, and it's not used in BaseExperiment

Reply:
Left it as it is now.

Reply:
But why not `stage=stage`? If it's not used right now it doesn't mean that it won't be used in the future

Reply:
Will make it `stage=stage`